layouts/partials/kubectl-reference-index.html

```diff
@@ -0,0 +1,48 @@
+{{- $parent := .Page -}}
+{{- $pages := (where .Site.Pages "Section" .Section).ByWeight -}}
+{{- $pages = (where $pages "Type" "!=" "search") -}}
+{{- $pages = (where $pages ".Parent" "!=" nil) -}}
+{{- $pages = (where $pages "Parent.File.UniqueID" "==" $parent.File.UniqueID) -}}
+
+<hr class="panel-line">
+<div id="kubectl-reference-index">
+
+    {{- range $index, $childPage := $pages -}} <!-- Iterate through all the section pages -->
+        {{- $sectionId := (replace $childPage.Title " " "-") | lower -}} <!-- Generate unique ID for each section page -->
+
+        <!-- Display the title of the child page as a header with an anchor link for easy navigation. -->
+        <h3 class="font-weight-bold mt-4 mb-1" id="{{ $sectionId }}"><a href=" {{ .RelPermalink }}">{{- $childPage.Title -}}</a></h3>
+
+        <!-- Details section for each section page -->
+        <p class="content-{{ $sectionId }}">
+            <details class="details-{{ $sectionId }}">
+                <summary class="card-header text-monospace summary-{{ $sectionId }}">{{- $childPage.Title -}}</summary>
+                <div class="border-bottom description-{{ $sectionId }}">
+                    {{- $content := $childPage.Content -}} <!-- Get child page content -->
+                    {{- $content = replaceRE "<h2 id=\"see-also\">.*?<\\/h2>\\s*<ul>([\\s\\S]*?)<\\/ul>" "" $content -}} <!-- Remove the 'See Also' section and its content -->
+                    {{- $content = replaceRE "<h2.*?>(.*?)<\\/h2>" "<h2>$1</h2>" $content -}} <!-- Remove 'id' attribute to eliminate the redundant deep link for sub-sections -->
+                    {{- $content = replaceRE "<table(\\s+[^>]*)?>" "<table class='table table-striped table-responsive'$1>" $content -}} <!-- Add 'class' attribute to table for similar look and feel to other tables on site -->
+                    {{- $content | safeHTML -}} <!-- Display modified content -->
+                </div>
+            </details>
```

Comment:
If we could, I'd prefer to change the generator code so that the HTML it generates is friendlier and requires less post-processing. Or even have the generator code do the whole thing.

Regular expressions over HTML are really hard to do robustly.